### DISCUSSION TITLE
How to get formatDistanceToNowStrict result into a specific format?

### DISCUSSION THREAD

**Discussion by niwatoridev** [js-fns/date-fns#3661] [Q&A] [answered]
I need to get the difference between two dates with a specific format (y + 'yrs' + m + 'mos'). I found date-fns and it works very cool, with the formatDistanceToNowStrict function it automatically gives me the difference between the date I put in and today's date, the problem is that it only gives me either the years (rounded) or the exact months, but I can't get the format I need. Does anyone know how or how else I can achieve this?

First image: Basically I need to get the difference between TODAY and the custom date (red box) and paint the result as in the blue box

Second image: functions with the date-fns library
![image](https://github.com/date-fns/date-fns/assets/88553099/cb7e6be9-90a3-4404-b691-1448ceed63ed)
![image](https://github.com/date-fns/date-fns/assets/88553099/23eb56b3-7695-4f35-ad7c-947f42b8844a)

**Comment by fturmel** [comment, marked as answer]
I'd recommend you leverage either [intervalToDuration](https://date-fns.org/docs/intervalToDuration) or [differenceInCalendarMonths](https://date-fns.org/docs/differenceInCalendarMonths) and create your own formatting function for your specific use case.

Also, don't input your date string in that non-standard format. It will fail on Webkit (Safari) and Firefox, and create an Invalid Date.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/Date#syntax
https://en.wikipedia.org/wiki/ISO_8601

Good luck!

**Reply by niwatoridev** [reply]
Awesome, thanks! I will try that. 

Regarding the formatting, if I understand correctly, my mistake is that I didnt add the 0 to the month, right? The correct format would be '2022-06' ?

**Reply by fturmel** [reply]
Yes, but a proper complete date is probably a better idea `2022-06-01`. Even better, use `new Date(2022, 5, 1)` or `new Date('2022-01-06T00:00:00')` so the input is in the local timezone instead of UTC and you won't get any surprises when using date-fns functions.
